src/Raven.Server/Rachis/RachisConsensus.cs

```diff
@@ -300,6 +305,8 @@ private set
 
         public Action<TransactionOperationContext, CommandBase> BeforeAppendToRaftLog;
 
+        public TaskCompletionSource<Task> OnSnapshotInstalledTask;
```

Comment:
Where is this being set to a value?

Reply:
You are correct. I've missed that.
Simplified to use only a `Task` instead of a `TCS` and added a test